app/services/layer-groups.js

```diff
@@ -1,107 +1,136 @@
 import Service from '@ember/service';
-import { A } from '@ember/array';
-import { copy } from 'ember-copy';
+import { set } from '@ember/object';
 
-const areIncomingLayerGroupsDefault = function(layerGroupsToShow, layerGroupsByDefault) {
-  const layersToShowAreDefault = layerGroupsToShow.every(
-    (layerGroupToShow, idx) => layerGroupToShow === layerGroupsByDefault[idx],
-  );
+export const isEqual = function(array1, array2) {
+  array1.sort();
+  array2.sort();
 
-  return layersToShowAreDefault;
+  return array1.every((value, index) => value === array2[index]);
+};
+
+const matchesIdentifierFor = function(layerGroup) {
+  return param => (param.id || param) === layerGroup.id;
+};
+
+export const applyQueryParamToLayerGroup = function(layerGroup, layerGroupIDsToShow) {
+  const paramForLayerGroup = layerGroupIDsToShow
+    .find(matchesIdentifierFor(layerGroup));
+
+  set(layerGroup, 'visible', !!paramForLayerGroup);
+
+  // handle layer group substate
+  if (paramForLayerGroup && paramForLayerGroup.selected) {
```

Comment:
If `.selected` is a boolean, then this block is equivalent to 
```
 if (paramForLayerGroup && paramForLayerGroup.selected) {
    set(layerGroup, 'selected', *true*);
}

```
Just noting it here in case it helps debugging down the road.

Reply:
Thanks @godfreyyeung, I actually don't know for sure haha. I believe it is a string, however - it's a specific layer-id.